File: src/Python/myPow.py
def my_pow(x, n):
    if n == 0:
        return 1
    if n < 0:
        x = 1 / x
        n = -n
    if n % 2 == 0:
        return my_pow(x * x, n // 2)
    else:
        return x * my_pow(x * x, n // 2)

# 测试
base = float(input("请输入底数 x："))
exponent = int(input("请输入指数 n："))
print(f"{base} 的 {exponent} 次方结果为：{my_pow(base, exponent)}")

File: src/Java/myPow.java
package Java;
import java.util.Scanner;

public class myPow {
        public static double myPow(double x, int n) {
            if (n == 0) {
                return 1.0;
            }
            if (n < 0) {
                x = 1.0 / x;
                n = -n;
            }
            if (n % 2 == 0) {
                return myPow(x * x, n / 2);
            } else {
                return x * myPow(x * x, n / 2);
            }
        }

        public static void main(String[] args) {
            Scanner scanner = new Scanner(System.in);
            System.out.print("请输入底数 x：");
            double base = scanner.nextDouble();
            System.out.print("请输入指数 n：");
            int exponent = scanner.nextInt();
            scanner.close();

            System.out.println(base + " 的 " + exponent + " 次方结果为：" + myPow(base, exponent));
        }
}
